Assets/HoloToolkit/Input/Scripts/KeywordManager.cs

```diff
@@ -84,26 +102,30 @@ void OnDestroy()
             }
         }
 
+        /// <summary>
+        /// Listens for key presses and calls the corrisponding keyword event
+        /// </summary>
         private void ProcessKeyBindings()
         {
-            foreach (var kvp in KeywordsAndResponses)
+            for (int i = 0; i < KeywordsAndResponses.Length; i++)
```

Comment:
That is true, but this loop is embedded in the update loop, and we'll need to iterate this array every frame. It might be best to keep it as a for loop to avoid any memory allocations during every frame.

This could potentially have a big impact of performance if we were to have a fairly large array, although, I think when it comes down to it, testing this and doing a study to see exactly what the performance impact would be might be best.